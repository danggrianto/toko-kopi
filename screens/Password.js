import React, { useState } from 'react'
import { TextInput, Button, View, Text, StyleSheet } from 'react-native'
import { CommonActions } from '@react-navigation/native'

import Colors from '../config/Colors'
import { firebase } from '../config/Firebase'

function Password({ navigation }) {
  const [email, setEmail] = useState('')
  const onPress = () => {
    firebase
      .auth()
      .sendPasswordResetEmail(email)
      .then(() => {
        alert('Please check your email...')
        navigation.dispatch(CommonActions.goBack())
      })
      .catch((e) => {
        console.error(e)
      })
  }
  return (
    <View style={styles.container}>
      <TextInput
        style={styles.inputBox}
        placeholder="enter your email address"
        placeholderTextColor="grey"
        autoCapitalize="none"
        onChangeText={(text) => setEmail(text)}
        value={email}
      />
      <Button title="Send Email" color={Colors.primary} onPress={onPress} />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputBox: {
    width: '85%',
    margin: 10,
    padding: 15,
    fontSize: 16,
    borderColor: Colors.outline,
    borderBottomWidth: 1,
    color: Colors.outline,
    fontWeight: 'bold',
    textAlign: 'center',
  },
})

export default Password
