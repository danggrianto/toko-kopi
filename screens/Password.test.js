import React from 'react'
import renderer from 'react-test-renderer'
import Password from './Password'

test('renders correctly', () => {
  const tree = renderer.create(<Password />).toJSON()
  expect(tree).toMatchSnapshot()
})
