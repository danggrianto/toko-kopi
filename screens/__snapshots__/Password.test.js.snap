// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="enter your email address"
    placeholderTextColor="grey"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#6D4539",
        "color": "#6D4539",
        "fontSize": 16,
        "fontWeight": "bold",
        "margin": 10,
        "padding": 15,
        "textAlign": "center",
        "width": "85%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#F8AF5E",
            },
          ]
        }
      >
        Send Email
      </Text>
    </View>
  </View>
</View>
`;
